docs: update readme (#1)
aquasecurity/vexhub: README.md

# VEX Hub

VEX Hub is a centralized repository that collects and manages [Vulnerability Exploitability eXchange (VEX)][vex] documents from various open-source software projects.
It serves as a comprehensive resource for vulnerability information, helping users and security tools to efficiently access and utilize VEX data across multiple projects and ecosystems.

VEX Hub automatically retrieves VEX documents from source repositories of registered projects.
By identifying the source repository from the registered [Package URL (PURL)][purl], VEX Hub copies and organizes the VEX documents, making them easily accessible to the wider community.

## Quick Start: Adding VEX Documents to VEX Hub

1. Create your VEX file
    1. VEX Hub supports [OpenVEX][openvex] or [CSAF][csaf] formats.

>     2. Store your VEX files in your source code repository under the `/.vex` directory at the root of your repository. More info [here](#discovery-of-vex-documents).

Reviewer: nit: It's a relative path. I prefer `.vex` or `.vex/`.

```suggestion
    2. Store your VEX files in your source code repository under the `.vex` directory at the root of your repository. More info [here](#discovery-of-vex-documents).
```

For example, GitHub says

>You must store workflow files in the `.github/workflows` directory of your repository.

https://docs.github.com/en/actions/writing-workflows/workflow-syntax-for-github-actions

That's it!
VEX Hub will automatically retrieve and process your VEX documents.

## VEX Crawling

[VEX Hub Crawler](vexhub-crawler) periodically crawls VEX documents of registered packages and maintains them in VEX Hub.
Registered packages are defined as a list of [PURLs][purl] in [a file](purl-list] that can be updated by anyone through Pull Requests.

For detailed information on PURL registration, supported ecosystems, and specific requirements, please refer to [VEX Hub Crawler][vexhub-crawler].

### Identifying Source Repositories

The packages file specifies only PURL of the package to be crawled, and VEX Hub Crawler automatically identifies the source code repository where the VEX file is located.
The method for identifying source repositories varies by ecosystem. For detailed information on how source repositories are identified and crawled for different package types, please refer to the [vex-crawler documentation](vexhub-crawler).

### Discovery of VEX Documents

After identifying the source code repository, VEX Hub automatically discovers VEX documents in it. The discovery process involves the following key points:

> - VEX documents are searched for in the `/.vex` directory at the root of the repository.

Reviewer: ditto

You should not trust VEX Hub any more than you trust the packages that you choose to use. 

VEX Hub does not have any responsibility or knowledge about the content of VEX documents it holds or the packages they describe. The VEX Hub maintainers do not review or vet the content of VEX documents, and the responsibility for the accuracy and correctness of the information lies with the originating package maintainers. We recommend that users manually triage suppressed vulnerabilities similarly to how they would triage any other vulnerability.

## Troubleshooting
If VEX documents in the source repository of a registered PURL are not being updated properly in VEX Hub, please check the following:

1. Ensure that the VEX documents are placed in the correct location (`.vex/` directory at the root of the repository).
2. Verify that the VEX documents use one of the supported file naming conventions.
3. Confirm that the PURL in the VEX document matches the registered PURL in VEX Hub.

> 4. Review the [vex-crawler documentation][vexhub-crawler] to ensure all specifications for crawling are met.

Reviewer: ```suggestion
4. Review the [vexhub-crawler documentation][vexhub-crawler] to ensure all specifications for crawling are met.
```

If issues persist after confirming these points, please open an issue in the VEX Hub repository for further assistance.

## Contributing
Contributions to improve VEX Hub are welcome!
Please note that the packages list, and associated code is maintained in the [VEX Hub Crawler repository][vexhub-crawler].
Pull Requests directly adding VEX documents to this repo are discouraged.

> This project and everyone participating in it is governed by the [Aquasecurity Code of Conduct][coc].

Reviewer: nit

```suggestion
This project and everyone participating in it is governed by the [Aqua Security Code of Conduct][coc].
```
